.github/workflows/pull-request-check.yml

```diff
@@ -71,7 +71,39 @@ jobs:
           cd tools/samm-cli
           unset JAVA_TOOL_OPTIONS
           mvn -B clean verify -Dmaven.wagon.httpconnectionManager.ttlSeconds=60
-          mvn -B verify -Pnative -Dmaven.wagon.httpconnectionManager.ttlSeconds=60
+        shell: bash
+
+      - name: Build native image
+        if: matrix.os != 'macos-13'
+        run: |
+          java --version
+          export MAVEN_OPTS="-Xmx4096m"
+          cd tools/samm-cli
+          mvn -B package -Pnative -Dmaven.wagon.httpconnectionManager.ttlSeconds=60
+        shell: bash
+
+      - name: Build native image (Mac)
+        if: matrix.os == 'macos-13'
+        run: |
+          mkdir samm-bundle
+          curl -Lo jre.tar.gz https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.3%2B9/OpenJDK21U-jre_x64_mac_hotspot_21.0.3_9.tar.gz
+          tar -xvf jre.tar.gz 
+          cp -r ./jdk-21.0.3+9-jre/Contents/Home ./samm-bundle/jre
+          cp tools/samm-cli/target/samm-cli-DEV-SNAPSHOT-for-native.jar ./samm-bundle/
```

Comment:
Don't use the `-for-native.jar` since it explictly does not include the runtime GraalVM APIs (because it confuses the GraalVM compiler). Only the regular jar without the suffix contains all necessary dependencies.